src/Serialization/Json/OkContentNormalizer.php

```diff
@@ -39,7 +46,43 @@ public function normalize($object, $format = null, array $context = array())
             $context['groups'] = $groups;
         }
 
-        return $this->decorated->normalize($object->getContent(), $format, $context);
+        // Simple object case
+        if (!$object->isCollection()) {
+            return $this->decorated->normalize($object->getContent(), $format, $context);
+        }
+
+        // Collection case
+        $content = $object->getContent();
+        $result = [];
+
+        // Pagination case
+        if ($content instanceof Pagerfanta) {
+            $result['meta'] = ['totalPages' => $content->getNbPages()];
+            $uri = $this->requestStack->getMasterRequest()->getUri();
+            $previousPage = $uri;
+            $nextPage = $uri;
+
+            if ($content->hasPreviousPage()) {
+                $previousPage = \preg_replace('/([?&])page=(\d+)/', '$1page=' . $content->getPreviousPage(), $uri);
+            }
+            if ($content->hasNextPage()) {
+                $nextPage = \preg_replace('/([?&])page=(\d+)/', '$1page=' . $content->getNextPage(), $uri);
+            }
+
+            $result['links'] = [
+                'prev' => $previousPage,
+                'next' => $nextPage,
```

Comment:
In case `$content->hasPreviousPage()` or `$content->hasNextPage()` are false these keys shouldn't be add to response. Current request's url isn't revelant

Reply:
Well, api-json say they should be here. So here they are 😇 .

Reply:
Mybad, you're absolutely right !

`Keys *MUST* either be omitted or have a null value to indicate that a particular link is unavailable.`